apps/integration/integration/containers/preexisting.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+import docker
+import requests
+import time
+from opensearchpy import OpenSearch
+from pathlib import Path
+from testcontainers.compose.compose import DockerCompose
+from typing import List
+
+
+def docker_compose(services: List[str] = []):
+    sycamore_root = Path(__file__).parent.parent.parent.parent.parent
+    print(sycamore_root)
+    if len(services) > 0:
+        return DockerCompose(sycamore_root, services=services)
+    else:
+        return DockerCompose(sycamore_root)
+
+
+@pytest.fixture(scope="session")
+def container_urls():
+    return {
+        "opensearch": ("localhost", 9200),
+        "rps": ("localhost", 2796),
+        "demo-ui": ("localhost", 3000),
+        "jupyter": ("localhost", 8888),
+    }
+
+
+@pytest.fixture(scope="session")
+def container_handles():
+    docker_client = docker.from_env()
+    jupyter = docker_client.containers.list(filters={"label": "com.docker.compose.service=jupyter"})[0]
```

Comment:
Just wondering if we can do containers.get("name")

Reply:
my main worry doing that is that docker compose likes to add a bunch of extra stuff to the name.
so the service 'jupyter' is named 'sycamore-jupyter-1'; if there are replicas I guess they'll get other numbers... This seemed safer